plugin/trino-tpcds/src/main/java/io/trino/plugin/tpcds/TpcdsMetadata.java

```diff
@@ -226,7 +225,6 @@ public static Type getTrinoType(ColumnType tpcdsType)
             case DECIMAL:
                 return createDecimalType(tpcdsType.getPrecision().get(), tpcdsType.getScale().get());
             case CHAR:
-                return createCharType(tpcdsType.getPrecision().get());
```

Comment:
The data types in tpcds connector are as per TPCDS spec and the ones in iceberg follow iceberg spec. You should handle the mismatch by casting to VARCHAR explicitly in your CTAS query rather than changing the code here